src/ui/Analytics/AnalyticsClient.ts

```diff
@@ -5,61 +5,109 @@ import { Promise } from 'es6-promise';
 import { PendingSearchEvent } from './PendingSearchEvent';
 
 /**
- * Describe an analytics client, that can log events or return information from the service.
+ * The IAnalyticsClient interface describes an analytics client that can log events to, or return information from the
+ * usage analytics service.
+ *
+ * See also the {@link Analytics} component.
  */
 export interface IAnalyticsClient {
   isContextual: boolean;
+
   /**
-   * Return false if there is no {@link Analytics} component in your page.
+   * Indicates whether there is an {@link Analytics} component in your page. Returns `true` if an Analytics component
+   * is present and `false` otherwise.
    */
   isActivated(): boolean;
   getCurrentEventCause(): string;
   getCurrentEventMeta(): { [key: string]: any };
+
   /**
-   * Log a search event on the service, using a cause and a meta object.<br/>
-   * Note that the event will be sent on the service when a query successfully return, not immediately after calling this method.<br/>
-   * Normally, this should be called using the following "format": <br/>
-   * this.usageAnalytics.logSearchEvent<SomeMeta>({name : 'foo', type : 'bar'}, <SomeMeta>{'key':'value'});<br/>
-   * this.queryController.executeQuery();<br/>
-   * This will queue up an analytics search event. Then the query is executed. The search event will be sent to the service when the query is successfully completed.<br/>
-   * @param actionCause
-   * @param meta Can be an empty object ( {} )
+   * Logs a Search event on the service, using an [AnalyticsActionCause]({@link IAnalyticsActionCause}) and a meta
+   * object.
+   *
+   * Note that the search event is sent to the service when a query successfully returns, not immediately after calling
+   * this method.
+   *
+   * Normally, you should call this method using the following "format":
+   *
+   * ```
+   * usageAnalytics.logSearchEvent<SomeMeta>({name: 'foo', type: 'bar'}, <SomeMeta>{'key':'value'});
+   * this.queryController.executeQuery();
```

Comment:
I would change this to something more generic, that simply explain that you need to call logSearchEvent before executing a query. It does not have to be the call to `queryController.executeQuery()`, explicitely.

It can be done with "Coveo.executeQuery()" for example.

I would also add a comments that talks about the error/information message that we display in the console when you do not do this.

You can test it by adding a CoveoAnalytics component in your page, and then programmatically trigger a query with Coveo.executeQuery, but omit to log a search event.

You should see an information message.